aead/src/dev.rs

```diff
@@ -1,74 +1,116 @@
 //! Development-related functionality
+use crate::{
+    Aead, AeadInOut, Nonce, Payload, Tag, TagPosition, array::typenum::Unsigned, inout::InOutBuf,
+};
 pub use blobby;
 
+/// Run AEAD test for the provided passing test vector
+pub fn run_pass_test<C: AeadInOut>(
+    cipher: &C,
+    nonce: &Nonce<C>,
+    aad: &[u8],
+    pt: &[u8],
+    ct: &[u8],
+) -> Result<(), &'static str> {
+    let res = cipher
+        .encrypt(nonce, Payload { aad, msg: pt })
+        .map_err(|_| "encryption failure")?;
+    if res != ct {
+        return Err("encrypted data is different from target ciphertext");
+    }
+
+    let res = cipher
+        .decrypt(nonce, Payload { aad, msg: ct })
+        .map_err(|_| "decryption failure")?;
+    if res != pt {
+        return Err("decrypted data is different from target plaintext");
+    }
+
+    let (ct, tag) = match C::TAG_POSITION {
+        TagPosition::Prefix => {
+            let (tag, ct) = ct.split_at(C::TagSize::USIZE);
+            (ct, tag)
+        }
+        TagPosition::Postfix => ct.split_at(pt.len()),
+    };
+    let tag: &Tag<C> = tag.try_into().expect("tag has correct length");
+
+    // Fill output buffer with "garbage" to test that its data does not get read during encryption
+    let mut buf: alloc::vec::Vec<u8> = (0..pt.len()).map(|i| i as u8).collect();
```

Comment:
not every consumer use the `aead::new_test` (at least deoxys does not) this is why the `MockBuffer` was made available in https://github.com/RustCrypto/traits/pull/1797#event-16931688965